lib/shopifyHMAC/index.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const crypto = require('crypto')
+
+const extractShopifySignature = ({ query }) => {
+  if (!query.hmac) {
+    throw new Error(
+      'Could not extract HMAC parameter: no HMAC parameter found in query.'
+    )
+  }
+  return query.hmac
+}
+
+const constructShopifySignature = (req, options) =>
+  calculateHmacParam({
+    algorithm: options.getAlgorithm(),
+    sharedSecret: options.sharedSecret,
+    message: getMessage({ ...req }),
+    digest: options.getDigest()
+  })
```

Comment:
can we de-structure here?

```suggestion
const constructShopifySignature = (req, { getAlgorithm, sharedSecret, getDigest }) =>
  calculateHmacParam({
    algorithm: getAlgorithm(),
    sharedSecret,
    message: getMessage({ ...req }),
    digest: getDigest()
  })
```